target/linux/ath79/dts/qca9558_compex_wpj558-16m.dts

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "compex,wpj558-16m", "qca,qca9558";
+	model = "Compex WPJ558 (16MB flash)";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_sig4;
+		led-failsafe = &led_sig4;
+		led-running = &led_sig4;
+		led-upgrade = &led_sig4;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		sig1 {
+			label = "red:sig1";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		sig2 {
+			label = "yellow:sig2";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		sig3 {
+			label = "green:sig3";
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		led_sig4: sig4 {
+			label = "green:sig4";
+			gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	beeper {
+		compatible = "gpio-beeper";
+		gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+	};
+};
+
+&spi {
+	status = "okay";
+
+	num-cs = <1>;
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <33000000>;
+		m25p,fast-read;
```

Comment:
Is this really necessary? Typically, fast-read is only required above 50 MHz.

Reply:
I don't understand what you mean? 33 MHz is only available for fast-read, and normal read should be lower?

Reply:
I double check the mx25l12805d datasheet and indeed the fast access time is at 50 MHz. Normal read is between 10KHz & 33 MHz.

# flash_speed /dev/mtd3 # 33 MHz               | # flash_speed /dev/mtd3 # 50 MHz
not NAND flash, assume page size is 512 bytes. | not NAND flash, assume page size is 512 bytes.
scanning for bad eraseblocks                   | scanning for bad eraseblocks
scanned 1 eraseblocks, 0 are bad               | scanned 1 eraseblocks, 0 are bad
testing eraseblock read speed                  | testing eraseblock read speed
eraseblock read speed is 438 KiB/s             | eraseblock read speed is 4000 KiB/s
testing page read speed                        | testing page read speed
page read speed is 435 KiB/s                   | page read speed is 3764 KiB/s
testing 2 page read speed                      | testing 2 page read speed
2 page read speed is 744 KiB/s                 | 2 page read speed is 4000 KiB/s
Testing 2x multi-block erase speed             | Testing 2x multi-block erase speed
2x multi-block erase speed is 224 KiB/s        | 2x multi-block erase speed is 228 KiB/s
Testing 4x multi-block erase speed             | Testing 4x multi-block erase speed
4x multi-block erase speed is 219 KiB/s        | 4x multi-block erase speed is 230 KiB/s
Testing 8x multi-block erase speed             | Testing 8x multi-block erase speed
8x multi-block erase speed is 223 KiB/s        | 8x multi-block erase speed is 227 KiB/s
Testing 16x multi-block erase speed            | Testing 16x multi-block erase speed
16x multi-block erase speed is 226 KiB/s       | 16x multi-block erase speed is 229 KiB/s
Testing 32x multi-block erase speed            | Testing 32x multi-block erase speed
32x multi-block erase speed is 220 KiB/s       | 32x multi-block erase speed is 228 KiB/s
Testing 64x multi-block erase speed            | Testing 64x multi-block erase speed
64x multi-block erase speed is 225 KiB/s       | 64x multi-block erase speed is 229 KiB/s
finished                                       | finished